spark-connect/client/src/main/scala-spark-master/io/delta/connect/tables/DeltaTable.scala

```diff
@@ -401,7 +401,12 @@ class DeltaTable private[tables](
     val extension = com.google.protobuf.Any.pack(relation)
     val sparkRelation = spark_proto.Relation.newBuilder().setExtension(extension).build()
     val result = sparkSession.newDataFrame(_.mergeFrom(sparkRelation)).collectResult()
-    sparkSession.createDataFrame(result.toArray.toSeq.asJava, result.schema)
+    val data = try {
```

Comment:
If we are going to do this, then you might as well use collect() instead of collectResult(). This is BTW not the first time I am seeing this

Reply:
Nvm you need the schema.